src/Tasks.UnitTests/MSBuild_Tests.cs

```diff
@@ -39,7 +39,6 @@ public void Dispose()
         /// throw a path too long exception
         /// </summary>
         [Fact]
-        [ActiveIssue("https://github.com/dotnet/msbuild/issues/4247")]
```

Comment:
One more question: 
It seems that the test fails when: "Fails when long paths are enabled."
Is there any chance to cover this by tests? long paths enabled/disabled ?

Reply:
This test fails since https://github.com/Microsoft/msbuild/blob/94e11e0a773bc8956caf128335433231bb06fed5/src/Build/BackEnd/Components/RequestBuilder/IntrinsicTasks/MSBuild.cs#L314.  The project file including multiple reductant "..\" and didn't get shorten there since long paths enabled in the function https://github.com/dotnet/msbuild/blob/7ca3c98fad986066bbf2802c863236b4a0f4e34a/src/Shared/FileUtilities.cs#L1192-L1200

https://github.com/Microsoft/msbuild/blob/94e11e0a773bc8956caf128335433231bb06fed5/src/Build/BackEnd/Components/RequestBuilder/IntrinsicTasks/MSBuild.cs#L314 is a bug.  I think this test could verify the bug.

Reply:
In such scenario the test does not cover the case when the long path are enabled. 
Currently we have static: 

```
for (int i = 0; i < 250; i++)
 {
           projectFile1 += "..\\";
 }
```
which does not guarantee to exceed the max path length .
and the resulting length does not depend on the actual max allowed path, but actual value of max allowed path could be 32,767 : https://learn.microsoft.com/en-us/windows/win32/fileio/maximum-file-path-limitation?tabs=registry

I would suggest to build the projectFile1 length based on current max file path length, to actually cover the case on different machines, without additional configuration, what do you think? 

Please let me know if I'm missing something.

Reply:
The maxpath is int.MaxValue if longpath is enabled.  We can't define a project file which length is longer than that.
https://github.com/dotnet/msbuild/blob/494047fc6f53e60048d23f0cc2350a042ba02526/src/Framework/NativeMethods.cs#L584-L595

Reply:
Thank you for the clarifications! yes agree, resolving the issue.

Reply:
Having that the test should be working on both solution it is not required for the project file to exceed max path.